rust/lance/src/dataset/scanner.rs

```diff
@@ -852,24 +852,24 @@ impl Scanner {
         } else {
             Arc::new(vec![])
         };
-        let knn_idx = indices.iter().find(|i| i.fields.contains(&column_id));
-        if let Some(index) = knn_idx {
+        if let Some(index) = indices.iter().find(|i| i.fields.contains(&column_id)) {
             // There is an index built for the column.
             // We will use the index.
-            if let Some(rf) = q.refine_factor {
-                if rf == 0 {
-                    return Err(Error::IO {
-                        message: "Refine factor can not be zero".to_string(),
-                        location: location!(),
-                    });
-                }
+            if q.refine_factor == Some(0) {
+                return Err(Error::IO {
+                    message: "Refine factor can not be zero".to_string(),
+                    location: location!(),
+                });
             }
 
-            let ann_node = self.ann(q, index, filter_plan).await?; // _distance, _rowid
+            // Find all deltas with the same index name.
+            let deltas = self.dataset.load_indices_by_name(&index.name).await?;
+            let ann_node = self.ann(q, &deltas, filter_plan).await?; // _distance, _rowid
```

Comment:
iiuc, this doubles ivf routing computation. For now probably okay, but we should have a ticket to dedup IVF calculation.

Reply:
Yea we can do the IVF first and fan out the PQ search later. Lemme take a look how much work that is.